sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/HashExpressionsSuite.scala

```diff
@@ -684,6 +684,21 @@ class HashExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     assert(murmur3HashPlan(wideRow).getInt(0) == murmursHashEval)
   }
 
+  test("SPARK-30633: Use Long seeds for xxHash") {
+    val literal = Literal.create(42L, LongType)
+    val seeds = Seq(
+      Long.MinValue,
+      Integer.MIN_VALUE.toLong - 1L,
+      0L,
+      Integer.MAX_VALUE.toLong + 1L,
+      Long.MaxValue
+    )
+
+    for (seed <- seeds) {
+      checkEvaluation(XxHash64(Seq(literal), seed), XxHash64(Seq(literal), seed).eval())
```

Comment:
To make it sure, can we compare the result with the `IntegerType` result?

Reply:
I also don't understand this request. What should we be comparing?

Reply:
Like do you get the same result with seed 100 vs 100L? You should, as "100" was already promoted to long in the generated code. Meh, if it's simple, might be worth a line to test, but shouldn't be a difference.